cli/lsp/config.rs

```diff
@@ -1265,13 +1279,11 @@ impl ConfigData {
     scope: &Arc<Url>,
     settings: &Settings,
     file_fetcher: &Arc<CliFileFetcher>,
+    http_client_provider: &Arc<HttpClientProvider>,
     // sync requirement is because the lsp requires sync
     deno_json_cache: &(dyn DenoJsonCache + Sync),
     pkg_json_cache: &(dyn PackageJsonCache + Sync),
     workspace_cache: &(dyn WorkspaceCache + Sync),
-    lockfile_package_info_provider: &Arc<
-      dyn deno_lockfile::NpmPackageInfoProvider + Send + Sync,
-    >,
```

Comment:
The previous code was not taking into account the npmrc for the current workspace directory when loading old lockfile versions. This needs to be constructed after the npmrc is resolved.

Reply:
Yikes, good catch